tools/options.c

```diff
@@ -333,8 +333,8 @@ static void nn_append_string (struct nn_parse_context *ctx,
 {
     struct nn_string_list *lst;
 
-    lst = (struct nn_string_list *)(
-        ((char *)ctx->target) + opt->offset);
+    lst = (struct nn_string_list *)(((char *)ctx->target) + opt->offset);
+    nn_assert (lst);
```

Comment:
This doesn't quite work, because the thing we must guarantee is that `lst` is indeed non-NULL, yet it's calculated from two other function parameters.